domain/status/state/modelstate.go

```diff
@@ -2048,7 +2048,14 @@ LEFT JOIN subnet AS sn ON ipa.subnet_uuid = sn.uuid
 		)
 
 		machineAddresses := addresses[s.UUID.String()]
-		matchedAddrs := machineAddresses.AllMatchingScope(corenetwork.ScopeMatchPublic)
+		matchedAddrs := make(corenetwork.SpaceAddresses, 0, len(machineAddresses))
+		for _, mAddr := range machineAddresses {
+			switch mAddr.Scope {
+			case corenetwork.ScopeMachineLocal, corenetwork.ScopeLinkLocal:
```

Comment:
I'd prefer if we only want the publicly scoped addresses that we check for a public address scope explicitly. This could match other undesirables if they exist.